backend/rpc-impl/src/main/java/org/sonarsource/sonarlint/core/rpc/impl/SonarLintRpcServerImpl.java

```diff
@@ -244,13 +257,33 @@ public CompletableFuture<Void> shutdown() {
         }
       }
       ThreadJobProcessorPool.getProcessors().forEach(JobProcessor::finish);
-      launcherFuture.cancel(true);
+      shutdownReaderAndWriter();
       return null;
     });
     executor.shutdown();
     return future;
   }
 
+  public void shutdownReaderAndWriter() {
+    messageReaderExecutor.shutdownNow();
+
+    //shutdown writer and disconnect from client asynchronously to make sure the client gets the response
+    var scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+    scheduledExecutorService.schedule(() -> {
+      messageWriterExecutor.shutdownNow();
+      disconnectFromClient();
```

Comment:
When I put it to the shutdown method, the process exits because the feature completes, and the client executes the code listening to the `onExit` instead of the code waiting for the shutdown response.

Because the process exits before sending the response, the client can't get the shutdown response.